opm/output/test_util/EclFilesComparator.cpp

```diff
@@ -0,0 +1,297 @@
+   */
+
+#include <opm/output/test_util/EclFilesComparator.hpp>
+#include <opm/common/ErrorMacros.hpp>
+
+#include <iostream>
+#include <algorithm>
+#include <ert/ecl/ecl_file.h>
+#include <ert/ecl/ecl_grid.h>
+
+bool ECLFilesComparator::keywordValidForComparing(const std::string keyword) const {
+    std::vector<std::string>::const_iterator it = find(keywords1.begin(), keywords1.end(), keyword);
+    if (it == keywords1.end()) {
+        std::cout << "Could not find keyword: " << keyword << std::endl;
+        OPM_THROW(std::runtime_error, "Keyword does not exist in first file.");
+    }
+    it = find(keywords2.begin(), keywords2.end(), keyword);
+    if (it == keywords2.end()) {
+        std::cout << "Could not find keyword: " << keyword << std::endl;
+        OPM_THROW(std::runtime_error, "Keyword does not exist in second file.");
+    }
+    ecl_kw_type* ecl_kw = ecl_file_iget_named_kw(ecl_file1, keyword.c_str(), 0);
+    ecl_type_enum ecl_kw_type = ecl_kw_get_type(ecl_kw);
+    switch (ecl_kw_type) {
+        case ECL_DOUBLE_TYPE:
+            return true;
+        case ECL_FLOAT_TYPE:
+            return true;
+        case ECL_CHAR_TYPE:
+            std::cout << "\nKeyword " << keyword << " is of type ECL_CHAR_TYPE, "
+                << "which is not supported for comparison." << std::endl;
+            return false;
+        case ECL_INT_TYPE:
+            std::cout << "\nKeyword " << keyword << " is of type ECL_INT_TYPE, "
+                << "which is not supported for comparison." << std::endl;
+            return false;
+        case ECL_BOOL_TYPE:
+            std::cout << "\nKeyword " << keyword << " is of type ECL_BOOL_TYPE, "
+                << "which is not supported for comparison." << std::endl;
+            return false;
+        case ECL_MESS_TYPE:
+            std::cout << "\nKeyword " << keyword << " is of type ECL_MESS_TYPE, "
+                << "which is not supported for comparison." << std::endl;
+            return false;
+        default:
+            std::cout << "Keyword " << keyword << "has undefined type." << std::endl;
+            return false;
+    }
+}
+
+
+
+void ECLFilesComparator::printResultsForKeyword(const std::string keyword) const {
+    const double absDeviationAverage = average(absDeviation);
+    const double relDeviationAverage = average(relDeviation);
+    std::cout << "\nKeyword "                    << keyword              << ":\n\n";
+    std::cout << "Average absolute deviation = " << absDeviationAverage  << std::endl;
+    std::cout << "Median absolute deviation  = " << median(absDeviation) << std::endl;
+    std::cout << "Average relative deviation = " << relDeviationAverage  << std::endl;
+    std::cout << "Median relative deviation  = " << median(relDeviation) << "\n\n";
+}
+
+
+
+void ECLFilesComparator::deviationsForReportStep(const std::string& keyword, int reportStep) {
+    ecl_kw_type* ecl_kw1               = ecl_file_iget_named_kw(ecl_file1, keyword.c_str(), reportStep);
```

Comment:
The variable here is not really `reportStep` but rather a flat index. A unified restart file can - and quite often *will* - contain "holes" in the series of report steps - e.g. the report steps available in the unified report from norne can be read off [here](https://github.com/OPM/opm-parser/blob/master/opm/parser/eclipse/IntegrationTests/IOConfigIntegrationTest.cpp#L53) as 0-5, 10, 19, 27, ...

The simplest "fix" is just to rename the variable to e.g. `occurence` or just plain `index`  - if you *really* mean report step the `ecl_file` code must be updated: https://github.com/Ensembles/ert/blob/master/devel/libecl/src/ecl_rstfile.c - I would start with just renaming.